src/classes/Households.cls

```diff
@@ -0,0 +1,201 @@
+                npo02__Household__c>(oldlist);
+    }
+
+    public void onBeforeUpdate() {
+        if (!isHouseholdTriggerContext() && isAdvancedHouseholdNamingEnabled()) {
+            setCustomNamingField();
+            setNameAndGreetingsToReplacementText();
+        }
+    }
+
+    public void onAfterUpdate() {
+        if (!isHouseholdTriggerContext() && isAdvancedHouseholdNamingEnabled()) {
+            renameHouseholdAccountsNeedingNameUpdates();
+        }
+    }
+
+    public void deleteEmptyHouseholds() {
+        delete emptyHouseholds().getRecords();
+    }
+
+    public Households emptyHouseholds() {
+        List<npo02__Household__c> emptyHouseholds = new List<npo02__Household__c>();
+        for (npo02__Household__c household : (List<npo02__Household__c>) getRecords()) {
+            if (household.npo02__Contacts__r.isEmpty()) {
+                emptyHouseholds.add(household);
+            }
+        }
+        return new Households(emptyHouseholds);
+    }
+
+    public Households nonEmptyHouseholds() {
+        List<npo02__Household__c> nonEmptyHouseholds = new List<npo02__Household__c>();
+        for (npo02__Household__c household : (List<npo02__Household__c>) getRecords()) {
+            if (!household.npo02__Contacts__r.isEmpty()) {
+                nonEmptyHouseholds.add(household);
+            }
+        }
+        return new Households(nonEmptyHouseholds);
+    }
+
+    public Map<Id, Integer> getHouseholdCountMap() {
+        Map<Id, Integer> householdCountMap = new Map<Id, Integer>();
+        for (npo02__Household__c household : (List<npo02__Household__c>) getRecords()) {
+            householdCountMap.put(household.Id, household.npo02__Contacts__r.size());
+        }
+        return householdCountMap;
+    }
+
+    public void calculateLegacyHouseholdRollups() {
+        Households nonEmptyHouseholds =
+                nonEmptyHouseholds();
+        if (nonEmptyHouseholds.ids().size() > 0) {
+            if (rollupsShouldRunSynchronously(nonEmptyHouseholds.ids())) {
+                RLLP_OppRollup rg = new RLLP_OppRollup();
+                rg.rollupHouseholds(getHouseholdMap(nonEmptyHouseholds.ids()));
+            } else {
+                RLLP_OppRollup.rollupHouseholdsFuture(nonEmptyHouseholds.ids());
+            }
+        }
+    }
+
+    public void updateHouseholdNameAndMemberCount() {
+        if (isFutureEligible()) {
+            HouseholdNamingService.updateHouseholdNameAndMemberCountAsynchronously(
+                    new List<Id>(ids()));
+        } else {
+            HouseholdNamingService householdNamingService = new HouseholdNamingService();
+            householdNamingService.updateHouseholdNameAndMemberCount(new List<Id>(ids()));
+        }
+    }
+
+    public Set<Id> ids() {
+        return getIdFieldValues(npo02__Household__c.Id);
+    }
+
+    private static List<npo02__Household__c> getHouseholdsById(Set<Id> ids) {
```

Comment:
Does this need to be static?

Reply:
In this configuration, yes, because it is called inside one of the constructors.  I've been hesitant to introduce Selector classes yet because I'd really like to have the full AEP library before doing that, but I introduced the very beginnings of a HouseholdSelector class here to avoid this, which should also help with future tests for this class.  Let me know what you think.

Reply:
Can the constructor not call an instance method?